source/docs/software/vision-processing/apriltag/apriltag-intro.rst

```diff
@@ -0,0 +1,237 @@
+
+Each image is searched for AprilTags using the algorithm described on this page. Using assumptions about how the camera's lense distorts the 3d world onto the 2d array of pixels in the camera, an estimate of the camera's position relative to the tag is calculated.
+
+The tag's ID is also decoded. from the image. Given each tag's ID, the position of the tag on the field can be looked up.
+
+Knowing the position of the tag on the field, and the position of the camera relative to the tag, the 3D geometry classes can be used to estimate the position of the camera on the field.
+
+If the camera's position on the robot is known, the robot's position on the field can also be estimated.
+
+These estimates can be incorporated into the WPILib pose estimation classes.
+
+2D to 3D Ambiguity
+------------------
+
+The process of translating the four known corners of the target in the image (two-dimensional) into a real-world position relative to the camera (three-dimensional) is inherently ambiguous. That is to say, there are multiple real-world positions that result in the target corners ending up in the same spot in the camera image.
+
+Humans can often use lighting or background clues to understand how objects are oriented in space. However, computers do not have this benefit. They can be tricked by similar-looking targets:
+
+.. image:: images/planar_ambiguity1_base.png
+   :alt: First optical illusion example of planar ambiguity
+   :width: 45 %
+
+.. image:: images/planar_ambiguity1.png
+   :alt: First optical illusion example of planar ambiguity
+   :width: 45 %
+
+.. image:: images/planar_ambiguity2.png
+   :alt: Second optical illusion example of planar ambiguity
+   :width: 45 %
+
+.. image:: images/planar_ambiguity3.png
+   :alt: Third optical illusion example of planar ambiguity
+   :width: 45 %
+
+Resolving which position is "correct" can be done in a few different ways:
+
+1. Use your odometry history from all sensors to pick the pose closest to where you expect the robot to be.
+2. Reject poses which are very unlikely (ex: outside the field perimiter, or up in the air)
+3. Ignore pose estimates which are very close together (and hard to differentiate)
+4. Use multiple cameras to look at the same target, such that at least one camera can generate a good pose estimate
+5. Look at many targets at once, using each to generate multiple pose estimates. Discard the outlying estimates, use the ones which are tightly clustered together.
+
+Adjustable Parameters
+---------------------
+
+``Decimation factor`` impacts how much the image is down-sampled before processing. Increasing it will increase detection speed, at the cost of not being able to see tags which are far away.
+
+``Blur`` applies smoothing to the input image to decrease noise, which increases speed when fitting quads to pixels, at the cost of precision. For most good cameras, this may be left at zero.
```

Comment:
A line about performance intensity? Maybe that should be left out since it is photon specific. just saying "good cameras" could leave it vague.

Reply:
Yea trying to keep the photon specifics out. I was also holding back on any recommendations about cameras, just cuz we're still waiting on seeing the on-field performance of different setups.